client/my-sites/media/main.jsx

```diff
@@ -44,14 +52,91 @@ module.exports = React.createClass( {
 		page( redirect );
 	},
 
+	openDetailsModal( item ) {
+		this.setState( { openedDetails: item } );
+	},
+
+	closeDetailsModal() {
+		this.setState( { openedDetails: null, editedItem: null } );
+	},
+
+	editImage() {
+		this.setState( { openedDetails: null, editedItem: this.state.openedDetails } );
+	},
+
+	onImageEditorCancel: function( imageEditorProps ) {
+		const {	resetAllImageEditorState } = imageEditorProps;
+		this.setState( { openedDetails: this.state.editedItem, editedItem: null } );
+
+		resetAllImageEditorState();
+	},
+	onImageEditorDone( error, blob, imageEditorProps ) {
+		if ( error ) {
+			this.onEditImageCancel( imageEditorProps );
+
+			return;
+		}
+
+		const {
+			fileName,
+			site,
+			ID,
+			resetAllImageEditorState
+		} = imageEditorProps;
+
+		const mimeType = MediaUtils.getMimeType( fileName );
+
+		const item = {
+			ID: ID,
+			media: {
+				fileName: fileName,
+				fileContents: blob,
+				mimeType: mimeType
+			}
+		};
+
+		MediaActions.update( site.ID, item, true );
+		resetAllImageEditorState();
+		this.setState( { openedDetails: null, editedItem: null } );
+	},
+
 	render: function() {
+		const site = this.props.sites.getSelectedSite();
 		return (
 			<div ref="container" className="main main-column media" role="main">
 				<SidebarNavigation />
+				{ ( this.state.editedItem || this.state.openedDetails ) &&
+					<Dialog
+						isVisible={ true }
+						additionalClassNames="editor-media-modal"
+						onClickOutside={ this.closeDetailsModal }
+						onClose={ this.closeDetailsModal }
+					>
+					{ this.state.openedDetails &&
+						<EditorMediaModalDetail
+							site={ site }
+							items={ [ this.state.openedDetails ] }
+							selectedIndex={ 0 }
+							onReturnToList={ this.closeDetailsModal }
+							onEditItem={ this.editImage }
+						/>
+					}
+					{ this.state.editedItem &&
+						<ImageEditor
+							siteId={ site && site.ID }
+							media={ this.state.editedItem }
+							onDone={ this.onImageEditorDone }
+							onCancel={ this.onImageEditorCancel }
+						/>
+					}
+					</Dialog>
+				}
 				<MediaLibrary
 					{ ...this.props }
 					onFilterChange={ this.onFilterChange }
-					site={ this.props.sites.getSelectedSite() || undefined }
```

Comment:
The fallback to `undefined` is because `SitesList#getSelectedSite` can return `false`, thereby causing a warning to be logged by unexpected prop type of `<MediaLibrary />`.